### PR TITLE
feat(table): zebra rows & columns

### PR DESCRIPTION
**Pull request by theJohnnyMe**
## **Describe pull-request**  
Adding "zebra" feature to table rows & columns

## **Issue Linking:**  
_Choose one of the following options_
- **Jira:** Add ticket number after `CDEP-`: [CDEP-3471](https://tegel.atlassian.net/browse/CDEP-3471)    
- **Jira:** Add ticket number after `CDEP-`: [CDEP-3472](https://tegel.atlassian.net/browse/CDEP-3472) 

## **How to test**  
1. Open preview link
2. Check Tables - > Zebra Mode
3. Check two different modes for rows and columns
4. Compare it with[ Figma design](https://www.figma.com/design/f0cqxOO7o859GNRkBKfEIx/Data-tables-exploration?node-id=1465-66217&m=dev)

## **Checklist before submission**
- [X] I have added unit tests for my changes (if applicable)
- [X] All existing tests pass
- [ ] I have updated the documentation (if applicable)

## **Suggested test steps**
- [x] Browser testing (Chrome, Safari, Firefox) 
- [ ] Keyboard operability
- [ ] Interactive elements have labels.
- [ ] Storybook controls
- [ ] Design/controls/props is aligned with other components 
- [x] Dark/light mode and variants 
- [ ] Input fields – values should be displayed properly 
- [ ] Events

## **Additional context**  

[CDEP-3471]: https://tegel.atlassian.net/browse/CDEP-3471?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CDEP-3472]: https://tegel.atlassian.net/browse/CDEP-3472?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/core/src/components/table/table-body/table-body.scss [hunk 1]
```diff
@@ -19,3 +19,75 @@ tds-table-body {
     transition: background-color 200ms ease;
   }
 }
+
+// Zebra mode: rows
```
Comment (timrombergjakobsson) [inline packages/core/src/components/table/table-body/table-body.scss:23]: Nice! If we want to we could DRY it up some more:

```
// Zebra mode: rows (odd and even)

// Combine odd and even row selectors
tds-table-body.tds-table--zebra-mode-rows-odd tds-table-body-row:nth-child(odd),
tds-table-body.tds-table--zebra-mode-rows-even tds-table-body-row:nth-child(even) {
  background-color: var(--tds-table-zebra-mode-backround);
}

tds-table-body.tds-table--zebra-mode-rows-odd tds-table-body-row:nth-child(odd):hover,
tds-table-body.tds-table--zebra-mode-rows-even tds-table-body-row:nth-child(even):hover {
  background-color: var(--tds-table-body-row-background-hover);
}

// Zebra mode: columns (odd and even)

// Combine odd and even column selectors
tds-table-body.tds-table--zebra-mode-columns-odd tds-body-cell:nth-child(odd),
tds-table-body.tds-table--zebra-mode-columns-even tds-body-cell:nth-child(even) {
  background-color: var(--tds-table-zebra-mode-backround);
}

tds-table-body.tds-table--zebra-mode-columns-odd tds-body-cell.tds-table__body-cell--hover:nth-child(odd),
tds-table-body.tds-table--zebra-mode-columns-even tds-body-cell.tds-table__body-cell--hover:nth-child(even) {
  background-color: var(--tds-table-body-cell-background-hover);
}

```
Just combining some of the repeating parts. But it comes down to personal preference :)

**Review by Lunkan89 — APPROVED** [review]
great work @theJohnnyMe

**Review by mistermalm — APPROVED** [review]
Looks good!

File: packages/core/src/components/table/table-body/table-body.scss (hunk 1 above)
Comment (theJohnnyMe) [reply packages/core/src/components/table/table-body/table-body.scss]: Done!

**Pull request merged**